book/src/external_state_transitions.md

```diff
@@ -0,0 +1,39 @@
+# External State Transitions (EIP-4788 & EIP-2935)
+
+Some Ethereum Improvement Proposals (EIPs) require state transitions that occur outside of the EVM execution engine, and thus outside of REVM itself. Notably, these are:
+
+- [EIP-4788: Beacon block root in the EVM](https://eips.ethereum.org/EIPS/eip-4788)
+- [EIP-2935: Add `blockHash` and `blockNumber` to the EVM](https://eips.ethereum.org/EIPS/eip-2935)
+
+## What are external state transitions?
+
+External state transitions refer to updates to the Ethereum state that are not performed by the EVM during transaction execution, but are instead performed by the client (or consensus layer) as part of block processing. These are typically required for EIPs that introduce new system contracts or require special state updates at block boundaries.
+
+## EIP-4788: Beacon block root in the EVM
+
+EIP-4788 requires that the root of each beacon chain block is committed to the execution layer and made available in the EVM via a special contract. This is achieved by the client (not the EVM) calling a system contract at a fixed address (`0x000F3df6D732807Ef1319fB7B8bB8522d0Beac02`) with the beacon root as input, at the start of each block. The contract maintains a ring buffer of recent roots.
+
+- The system call is performed by the client, not by EVM transaction execution.
+- If the contract does not exist, the call fails silently.
+- See [EIP-4788](https://eips.ethereum.org/EIPS/eip-4788) for full details.
+- Example implementation in Reth: [reth#4457](https://github.com/paradigmxyz/reth/pull/4457)
+
+## EIP-2935: Add blockHash and blockNumber to the EVM
+
+EIP-2935 introduces a system contract that stores recent block hashes, allowing contracts to query them. The client is responsible for updating this contract at each block, by calling a system contract at a fixed address (`0x0000F90827F1C53a10cb7A02335B175320002935`) with the new block hash.
+
+- The system call is performed by the client, not by EVM transaction execution.
+- See [EIP-2935](https://eips.ethereum.org/EIPS/eip-2935) for full details.
+- Example implementation in Reth: [reth#7818](https://github.com/paradigmxyz/reth/pull/7818)
+
+## How does this affect REVM users?
+
+- REVM itself does not perform these external state transitions. It expects the state to be updated by the client or test harness before EVM execution.
```

Comment:
To perform this block state transition system call should be used that Revm supports.